javascripts/discourse/components/custom-admin-menu.gjs

```diff
@@ -0,0 +1,87 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { on } from "@ember/modifier";
+import { LinkTo } from "@ember/routing";
+import { service } from "@ember/service";
+import icon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+import DMenu from "float-kit/components/d-menu";
+
+export default class CustomAdminMenu extends Component {
+  @service currentUser;
+  @service store;
+  @service pmTopicTrackingState;
+
+  @tracked botUser = null;
+
+  constructor() {
+    super(...arguments);
+    this.loadBotUser();
+  }
+
+  async loadBotUser() {
+    this.botUser = await this.store.find("user", "gpt-4o");
+  }
+
+  get modHasMessages() {
+    this.pmTopicTrackingState.activeGroup = "moderators";
+    return this.pmTopicTrackingState?.newIncoming?.length;
```

Comment:
You are using safe nav `?.` here but not above. I don't think it's needed for the `this.pmTopicTrackingState?.` since that service should always exist